### PR TITLE
builtin/k8s: Surface pod failures on deploy

### PR DESCRIPTION
**Pull request by briancain**
Prior to this commit, if a pod failed to pull an image, the error
wouldn't surface to the user and it would just look like the deployment
was hanging. This commit fixes that by using the `ui` to surface the
known pod failure of not being able to pull a container image.

### CODE REVIEW

**Comment by briancain** [comment]
Just a quick gif of what this change looks like. It doesn't end up repeating itself during the poll loop and only prints once. It'll spin there for ~10 minutes until it fails with a single warning.

Peek 2021-02-18 09-59

File: builtin/k8s/platform.go [hunk 1]
```diff
@@ -429,7 +429,9 @@ func (p *Platform) Deploy(
 				}
 
 				if cs.State.Waiting != nil {
-					if cs.State.Waiting.Reason == "ImagePullBackOff" {
+					// TODO: handle other pod failures here
+					if cs.State.Waiting.Reason == "ImagePullBackOff" ||
+						cs.State.Waiting.Reason == "ErrImagePull" {
```
Comment (evanphx) [inline builtin/k8s/platform.go:434]: Didn't the old version warn in the same way but actually catch more conditions? I'm confused.

**Review by evanphx — CHANGES_REQUESTED** [review]

File: builtin/k8s/platform.go (hunk 1 above)
Comment (briancain) [reply builtin/k8s/platform.go:434]: The old version only looked at `"ImagePullBackOff"`, which happens after `"ErrImagePull"`. I'm not sure if they are exclusively similar, but to me the error we're actually trying to handle here is when it fails to pull the image. I'm not sure what you mean by catching more conditions on the old version, unless there's a change prior to this PR you're talking about :smile:

Comment (evanphx) [reply builtin/k8s/platform.go:434]: oh I can't read. Nevermind!

**Review by evanphx — APPROVED** [review]

**Review by krantzinator — APPROVED** [review]

**Pull request merged**